lib/private/Share20/DefaultShareProvider.php

```diff
@@ -671,29 +671,39 @@ public function getSharesInFolder($userId, Folder $node, $reshares) {
 		}
 
 		// todo? maybe get these from the oc_mounts table
-		$childMountNodes = array_filter($node->getDirectoryListing(), function (Node $node) {
+		$childMountNodes = array_filter($node->getDirectoryListing(), function (Node $node): bool {
 			return $node->getInternalPath() === '';
 		});
-		$childMountRootIds = array_map(function (Node $node) {
+		$childMountRootIds = array_map(function (Node $node): int {
 			return $node->getId();
 		}, $childMountNodes);
 
 		$qb->innerJoin('s', 'filecache', 'f', $qb->expr()->eq('s.file_source', 'f.fileid'));
 		$qb->andWhere(
 			$qb->expr()->orX(
 				$qb->expr()->eq('f.parent', $qb->createNamedParameter($node->getId())),
-				$qb->expr()->in('f.fileid', $qb->createNamedParameter($childMountRootIds, IQueryBuilder::PARAM_INT_ARRAY))
+				$qb->expr()->in('f.fileid', $qb->createParameter('chunk'))
 			)
 		);
 
 		$qb->orderBy('id');
 
-		$cursor = $qb->execute();
 		$shares = [];
-		while ($data = $cursor->fetch()) {
-			$shares[$data['fileid']][] = $this->createShare($data);
+
+		if (count($childMountRootIds) === 0) {
+			$chunks = [[]];
+		} else {
+			$chunks = array_chunk($childMountRootIds, 1000);
+		}
```

Comment:
Don't know, might be more readable :)

```suggestion
		$chunks = array_chunk($childMountRootIds, 1000);

		// Force the request to be run when there is 0 mount.
		if (count($chunks) === 0) {
			$chunks = [[]];
		}
```